### PR TITLE
Closes #36 update readme

### LINKED ISSUE #36: Update README
- Naming Conventions: https://github.com/pharmaverse/pharmaversesdtm/issues/29
-               `[Github repository](https://github.com/pharmaverse/admiral.test/tree/main/data)` should refer to the page created from `R/data.R` in admiraldata.

_Originally posted by @bundfussr in https://github.com/pharmaverse/admiraldev/pull/293#discussion_r1246689482_
            
-               I think we need to clarify what this means. This process should be followed if the new dataset is created by a program. If it is taken from an external source like the CDISC pilot data no program is required.

_Originally posted by @bundfussr in https://github.com/pharmaverse/admiraldev/pull/293#discussion_r1246694927_
            
- ...

### PR DESCRIPTION
**Pull request by kaz462**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by kaz462** [comment]
- README is updated based on the most recent folder structure and naming conventions discussions 

- For CODEOWNERS, will create a full list and update it another time https://github.com/pharmaverse/pharmaversesdtm/issues/43

File: README.md [hunk 1]
```diff
@@ -28,39 +26,44 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
 ## Adding New SDTM Datasets
 
-* Create a program in the `dev/` folder, named `<name>.R`, where `<name>` is the SDTM domain name, (e.g. `rs.R`), to generate the test data and output `<name>.rda` to the `data/` folder. Use CDISC pilot data such as `dm` as input in this program in order to create realistic synthetic data that remains consistent with other domains. Note that __no personal data should be used__ as part of this package, even if anonymized.
+* Create a program in the `data-raw/` folder, named `<name>.R`, where `<name>` should follow the [naming convention](#naming), to generate the test data and output `<name>.rda` to the `data/` folder.
+    * Use CDISC pilot data such as `dm` as input in this program in order to create realistic synthetic data that remains consistent with other domains (not mandatory).
+    * Note that **no personal data should be used** as part of this package, even if anonymized.
 * Run the program.
 * Reflect this update, by specifying `<name>` in `R/data.R`.
 * Run `devtools::document()` in order to update `NAMESPACE` and update the `.Rd` files in `man/`.
+* Add your GitHub handle to `.github/CODEOWNERS`
```
Comment (bundfussr) [inline README.md:63]: I would add that `NEWS.md` must be updated.

File: README.md [hunk 2]
```diff
@@ -28,39 +26,44 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
 
-* Programs that generate test data are stored in the `dev/` folder.
-* Each of these programs is written as a standalone R script: if any packages need to be loaded for a given program, then call `library()` at the start of the program (but please do __not__ call `library(pharmaversesdtm)`).
+* Programs that generate test data are stored in the `data-raw/` folder.
+* Each of these programs is written as a standalone R script: if any packages need to be loaded for a given program, then call `library()` at the start of the program (but please do **not** call `library(pharmaversesdtm)`).
 * Most of the packages that you are likely to need will already be specified in the `renv.lock` file, so they will already be installed if you have been keeping in sync--you can check this by entering `renv::status()` in the Console. However, you may also wish to install `{metatools}` and `{ggplot2}`, which are currently not specified in the `renv.lock` file. If you feel that you need to install any other packages in addition to those just mentioned, then please tag `@pharmaverse/admiral` to discuss with the development core team.
```
[the remark this answers is not in the document]
Comment (kaz462) [reply README.md:45]: Those are directly from `admiral.test`. I'm not sure why ggplot2 was used, maybe the author wanted to check if the simulated data looks reasonable? 
I'm fine with removing it or leaving it as is. @manciniedoardo what do you think?

File: README.md [hunk 3]
```diff
@@ -28,39 +26,44 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
-* The programs in `dev/` are stored within the `{pharmaversesdtm}` GitHub repository, but they are __not__ part of the `{pharmaversesdtm}` package--the `dev/` folder is specified in `.Rbuildignore`.
-* When you run a program that is in the `dev/` folder, you generate a dataset that is written to the `data/` folder, which will become part of the `{pharmaversesdtm}` package.
-* The names of test datasets are specified in `R/data.R`, for the purpose of generating documentation in the `man/` folder.
+* When you have created a program in the `data-raw/` folder, you need to run it as a standalone R script, in order to generate a test dataset that will become part of the `{pharmaversesdtm}` package, but you do not need to build the package.
+* Following [best practice](https://r-pkgs.org/data.html#sec-data-data), each dataset is stored as a `.rda` file whose name is consistent with the name of the dataset, e.g., dataset `xx` is stored as `xx.rda`
+    * `save(xx, file = "data/xx.rda", compress = "bzip2")`
+    * or an easier way to achieve this is to use `usethis::use_data()`
```
[the remark this answers is not in the document]
Comment (kaz462) [reply README.md:49]: sounds good! updated and added a comment in the [Clean up scripts issue](https://github.com/pharmaverse/pharmaversesdtm/issues/39)

File: README.md [hunk 4]
```diff
@@ -28,39 +26,44 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
+Some of the test datasets has been sourced from the [CDISC pilot project](https://github.com/cdisc-org/sdtm-adam-pilot-project), while other datasets have been constructed ad-hoc by the admiral team. Please check the [Github repository](https://github.com/pharmaverse/pharmaversesdtm/blob/main/data) for detailed information regarding the source of specific datasets.
+
+# Naming Conventions {#naming}
 
-# Naming Conventions
 * Datasets that are TA-agnostic: same as SDTM domain name (e.g., `dm`, `rs`).
-* Datasets that are TA-specific: prefix the domain name with the TA (e.g., `onco_rs`, `ophtha_oe`).
+* Datasets that are TA-specific: domain_TA_others, others go from broader categories to more specific ones (e.g., `oe_ophtha`, `rs_onco`, `rs_onco_irecist`).
 
 **Note**: *If an SDTM domain is used by multiple TAs, `{pharmaversesdtm}` may provide multiple versions of the corresponding test dataset. For instance, the package contains `ex` and `ophtha_ex` as the latter contains ophthalmology-specific variables such as `EXLAT` and `EXLOC`, and `EXROUTE` is exchanged for a plausible ophthalmology value.*
```
[the remark this answers is not in the document]
Comment (kaz462) [reply README.md]: good catch, thanks for the review!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply README.md:45]: yes please remove.

File: README.md [hunk 5]
```diff
@@ -28,39 +26,45 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
 
-* Create a program in the `dev/` folder, named `<name>.R`, where `<name>` is the SDTM domain name, (e.g. `rs.R`), to generate the test data and output `<name>.rda` to the `data/` folder. Use CDISC pilot data such as `dm` as input in this program in order to create realistic synthetic data that remains consistent with other domains. Note that __no personal data should be used__ as part of this package, even if anonymized.
+* Create a program in the `data-raw/` folder, named `<name>.R`, where `<name>` should follow the [naming convention](#naming), to generate the test data and output `<name>.rda` to the `data/` folder.
+    * Use CDISC pilot data such as `dm` as input in this program in order to create realistic synthetic data that remains consistent with other domains (not mandatory).
+    * Note that **no personal data should be used** as part of this package, even if anonymized.
 * Run the program.
 * Reflect this update, by specifying `<name>` in `R/data.R`.
 * Run `devtools::document()` in order to update `NAMESPACE` and update the `.Rd` files in `man/`.
+* Add your GitHub handle to `.github/CODEOWNERS`
+* Update `NEWS.md`
```
Comment (manciniedoardo) [inline README.md:61]: full stops are missing at the end of the last two bullet points.

File: README.md [hunk 6]
```diff
@@ -28,39 +26,45 @@ remotes::install_github("pharmaverse/pharmaversesdtm", ref = "devel")
...
 ## Updating Existing SDTM Datasets
 
-* Rename the source dataset as `raw_<name>`, where `<name>` is the SDTM domain name (e.g. rename `ds` to `raw_ds`), and then save it to the `data/` folder as `raw_<name>.rda` (e.g. `save(raw_ds, file = "data/raw_ds.rda")`).
-* Create a program in the `dev/` folder, named `update_<name>.R`, to load `raw_<name>.rda`, make the updates, and output `<name>.rda` to the `data/` folder.
-* Run the program.
-* Reflect this update, by specifying both `raw_<name>` and `<name>` in `R/data.R`.
+* Locate the existing program `<name>.R` in the `data-raw/` folder, update it accordingly. 
+* Run the program, and output updated `<name>.rda` to the `data/` folder.
 * Run `devtools::document()` in order to update `NAMESPACE` and update the `.Rd` files in `man/`.
+* Add your GitHub handle to `.github/CODEOWNERS`
```
Comment (manciniedoardo) [inline README.md:68]: full stops are missing at the end of the last two bullet points.

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
lgtm, just a few final minor points

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaz462) [reply README.md]: removed

File: README.md (hunk 6 above)
Comment (kaz462) [reply README.md]: updated

**Review by manciniedoardo — APPROVED** [review]

**Comment by kaz462** [comment]
@manciniedoardo @bundfussr Thanks for the review!
